pkg/api/validation/validation_test.go

```diff
@@ -3597,176 +3619,192 @@ func TestValidatePod(t *testing.T) {
-								}]
-							},
-							"namespaces": ["INVALID_NAMESPACE"],
-							"topologyKey": "region"
-						}
-					}]}}`,
-				},
 			},
-			Spec: validPodSpec,
-		},
-		"invalid labelOperator in preferredDuringSchedulingIgnoredDuringExecution in podantiaffinity annotations, labelOperator should be proper": {
-			ObjectMeta: api.ObjectMeta{
-				Name:      "123",
-				Namespace: "ns",
-				Annotations: map[string]string{
-					api.AffinityAnnotationKey: `
-					{"podAntiAffinity": {"preferredDuringSchedulingIgnoredDuringExecution": [{
-						"weight": 10,
-						"podAffinityTerm":
-						{
-							"labelSelector": {
-								"matchExpressions": [{
-									"key": "key2",
-									"operator": "WrongOp",
-									"values": ["value1", "value2"]
-								}]
+			Spec: api.PodSpec{
+				Containers:    []api.Container{{Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent"}},
+				RestartPolicy: api.RestartPolicyAlways,
+				DNSPolicy:     api.DNSClusterFirst,
+				Affinity: &api.Affinity{
+					PodAffinity: &api.PodAffinity{
+						PreferredDuringSchedulingIgnoredDuringExecution: []api.WeightedPodAffinityTerm{
+							{
+								Weight: 10,
+								PodAffinityTerm: api.PodAffinityTerm{
+									LabelSelector: &metav1.LabelSelector{
+										MatchExpressions: []metav1.LabelSelectorRequirement{
+											{
+												Key:      "key2",
+												Operator: metav1.LabelSelectorOpExists,
+											},
+										},
+									},
+									Namespaces:  []string{"INVALID_NAMESPACE"},
+									TopologyKey: "region",
+								},
 							},
-							"namespaces": ["ns"],
-							"topologyKey": "region"
-						}
-					}]}}`,
+						},
+					},
 				},
 			},
-			Spec: validPodSpec,
 		},
 		"invalid pod affinity, empty topologyKey is not allowed for hard pod affinity": {
 			ObjectMeta: api.ObjectMeta{
 				Name:      "123",
 				Namespace: "ns",
-				Annotations: map[string]string{
-					api.AffinityAnnotationKey: `
-					{"podAffinity": {"requiredDuringSchedulingIgnoredDuringExecution": [{
-						"weight": 10,
-						"podAffinityTerm":
-						{
-							"labelSelector": {
-								"matchExpressions": [{
-									"key": "key2",
-									"operator": "In",
-									"values": ["value1", "value2"]
-								}]
+			},
+			Spec: api.PodSpec{
+				Containers:    []api.Container{{Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent"}},
+				RestartPolicy: api.RestartPolicyAlways,
+				DNSPolicy:     api.DNSClusterFirst,
+				Affinity: &api.Affinity{
+					PodAffinity: &api.PodAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: []api.PodAffinityTerm{
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchExpressions: []metav1.LabelSelectorRequirement{
+										{
+											Key:      "key2",
+											Operator: metav1.LabelSelectorOpIn,
+											Values:   []string{"value1", "value2"},
+										},
+									},
+								},
+								Namespaces:  []string{"ns"},
+								TopologyKey: "",
```

Comment:
Original test has an empty value for topologyKey.  I would think it shouldn't matter if it is there and empty vs not there at all.  I will test and if there's no difference just remove it